### PR TITLE
Fix webpack deprecations (fix #1126)

### LINKED ISSUE #1126: Deprecation Warning on Webpack 5
### Expected Behaviour
Clean output without deprecation warning.

### Actual Behaviour

After webpack update (from v4 to v5) I'm getting deprecation warnings in my console.

```
(node:7361) [DEP_WEBPACK_DEPRECATION_ARRAY_TO_SET] DeprecationWarning: Compilation.modules was changed from Array to Set (using Array method 'reduce' is deprecated)

(node:41349) [DEP_WEBPACK_MODULE_ERRORS] DeprecationWarning: Module.errors was removed (use getErrors instead)
```

### Steps to Reproduce the Problem

Using tracing:

```sh
node --trace-deprecation node_modules/webpack/bin/webpack.js
```

Output:

```
(node:41650) [DEP_WEBPACK_DEPRECATION_ARRAY_TO_SET] DeprecationWarning: Compilation.modules was changed from Array to Set (using Array method 'reduce' is deprecated)
    at Set.set.<computed> [as reduce] (/Users/.../myproject/node_modules/webpack/lib/util/deprecation.js:78:4)
    at determineModules (/Users/.../myproject/node_modules/ts-loader/dist/after-compile.js:58:32)
```

```
(node:41650) [DEP_WEBPACK_MODULE_ERRORS] DeprecationWarning: Module.errors was removed (use getErrors instead)
    at /Users/.../myproject/node_modules/ts-loader/dist/after-compile.js:140:45
    at Array.forEach (<anonymous>)
```

### Environment

```
Node.js v12.16.2
webpack@5.0.0-beta.16
ts-loader@7.0.5
```

### PR DESCRIPTION
**Pull request by g-plane**
This PR should remove some of webpack deprecation warnings.

### CODE REVIEW

**Comment by johnnyreilly** [comment]
@sheetalkamat has a PR which drops support for older versions of TypeScript here: https://github.com/TypeStrong/ts-loader/pull/1136

When that's merged we can take a look at your PR - it only seems to be failing on older versions of TypeScript.

**Comment by g-plane** [comment]
It seems that these are weird failures, and I'm not sure whether upgrading webpack will fix these or not.

**Comment by johnnyreilly** [comment]
Very strange....

File: src/after-compile.ts [hunk 1]
```diff
@@ -63,6 +61,7 @@ export function makeAfterCompile(
 
     const filesWithErrors: TSFiles = new Map();
     provideErrorsToWebpack(
+      loader,
```
Comment (sheetalkamat) [inline src/after-compile.ts:64]: why do you need to pass this as instance.loader is there?

**Comment by g-plane** [comment]
I've confirmed that upgrading webpack will solve the failures in CI, as those failures also can be reproduced on my local computer. Can I open another PR to upgrade webpack? If not, we may need to add some hacky code to check the loader context.

**Comment by johnnyreilly** [comment]
Yeah sure - I'd like to decouple the upgrade of webpack from this change if that's okay. So upgrade webpack first and loop back to this next.

**Comment by g-plane** [comment]
PR for upgrading webpack is created: #1140 .

**Comment by g-plane** [comment]
I've marked this PR as ready for review, however I don't know how to address the problems at AppVeyor.

**Comment by johnnyreilly** [comment]
AppVeyor has had a bad day - let's see if it fixes up now.

**Comment by g-plane** [comment]
Maybe we can migrate to GitHub Actions in the future.

**Comment by johnnyreilly** [comment]
We're already using GitHub actions, but it doesn't cover the use case that AppVeyor provides; namely running CI on Windows.

Maybe Azure pipelines has something to offer - I haven't taken a look in a while. I did start looking here though:

https://github.com/TypeStrong/ts-loader/pull/856

**Comment by g-plane** [comment]
I will. By the way, GitHub Actions supports Linux, Windows and macOS.

**Comment by johnnyreilly** [comment]
Really? Did not know that! That said, I never managed to get comparison tests passing with GitHub actions. But that was the early days

**Comment by g-plane** [comment]
Maybe we can try it after this PR.

**Comment by g-plane** [comment]
I've updated `package.json` and changelog. Is this OK?

**Comment by johnnyreilly** [comment]
Looks good! Let's wait for Travis to get happy then we can merge

**Comment by johnnyreilly** [comment]
Right now I want AppVeyor to die

**Comment by g-plane** [comment]
All passed now.

**Comment by johnnyreilly** [comment]
Shipping https://github.com/TypeStrong/ts-loader/releases/tag/v8.0.1 - thanks for your work!

**Pull request merged**